### ISSUE TITLE
Need ability to read config for multi domain and authenticate on them

### ISSUE THREAD

**Issue by candleshine** [sakuraapi/auth-audience#13]

**Comment by candleshine** [comment]
format would look like this
```
authentication": {
    "jwt": {
      "domainedAudiences": {
        "default": {
          "default.server": "keyfordefault"
        },
        "domain2": {
          "domain2.server": "keyfordomain2"
        }
      },
      "issuer": "profile.illuminations.bible"
    }
  }
```

**Comment by etsuo** [comment]
@candleshine, currently the config looks like this:
```
modules.export = {
  authentication: {
    jwt: {
      audience: "audience.somedomain.somewhere",
      issuer: "issuer.somedomain.somewhere",
      key: "12345678901234567890123456789012"
    }
  }
};
```
We need to be able to support the following scenarios:

1. having multiple issuers (#16 might resolve this partially, but it assumes both issuers would have the same private key, which is wrong)
    ```
     jwt: {
            audience: "audience.somedomain.somewhere",
            issuers: { 
                 "issuer1.somedomain.somewhere": "key123" 
                 "issuer2.somedomain.somewhere": "key456" 
            }
     }
    ```
1. being able to pose as multiple audiences (i.e., a single server could be a different audience from the perspective of different issuers, or the same issuer acting as the authority for multiple domains)
    ```
    jwt: {
        audiences: { 
                "audience1.somedomain.somewhere": {
                        issuer: "issuer1.somedomain.somewhere"
                        key: "123"
                },
                "audience2.somedomain.somewhere": {
                        issuer: "issuer2.somedomain.somewhere"
                        key: "456"
                }
        }
    }
    ```
1. being able to handle multiple domains (i.e., depending on the domain in the JWT, determining if the audience is supported and if the JWT signs correctly with the secret key for that audience given the domain)
    ```
    jwt: {
        domainedAudiences: {
            "field": {
                audience: "audience1.somedomain.somewhere",
                issuer: "issuer1.somedomain.somewhere",
                key: "123"
            },
            "default": {
                audience: "audience2.somedomain.somewhere",
                issuer: "issuer2.somedomain.somewhere",
                key: "456"
            }
        }
    }
    ```

The implementation needs to support the existing configuration above (the simple case of the audience server only acting as one audience and only having a single issuer).

It also needs to support the scenarios listed above.

Right now, you are blocked by the third scenario, so perhaps focus on that. But we need 1 & 2 as well.

**Comment by candleshine** [comment]
This issue will cover only scenario number 3 above.  A new ticket will handle the other scenarios.

**Comment by etsuo** [comment]
releasing on 0.4.0
